### PR TITLE
Feature/Error Handling

### PR DESCRIPTION
**Pull request by Melizzo**
# PR Description
Please include a summary of the change:
Added functionality to the buttons on login - buttons are disabled if the input is empty, and added in trim() that removes any white spaces that the user may have entered in.  

Commit Emojis:
:tada: Initial commit
🔮  JavaScript components

If it is a new feature, describe the feature and the iteration of the project spec it addresses.

Note any questions or struggles as well.
Attempted refactor to add message for user clicking on under 21 to be a message on the DOM, instead of an alert.  However, placed on refactor, as a whole new component, and state would need to be created.   I chose to continue to focus to create a functioning app and tests.  User impact is minimal. 

## Type of change
Please delete options that are not relevant.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Tests (non-breaking change which adds testing)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [ ] My code follows the Turing style guidelines
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/Login/Login.tsx [hunk 1]
```diff
@@ -1,65 +1,75 @@
...
+import "./Login.scss";
 
 export interface LoginProps {
-	username: string;
-	setUsername: any;
-	loggedIn: boolean;
-	setLoggedIn: any;
-}
+  username: string;
+  setUsername: Function;
```
Comment (Melizzo) [inline src/Login/Login.tsx:7]: Added in TS type as Function for setUsername & setLoggedIn

File: src/Login/Login.tsx [hunk 2]
```diff
@@ -1,65 +1,75 @@
...
+  setLoggedIn: Function;
+} //generic ready if we need to safe type any props passed in
 
-const Login: React.SFC<LoginProps> = (props) => {	
-	const verifyUser = () => {
-		props.setLoggedIn(!props.loggedIn);
-	};
+const Login: React.SFC<LoginProps> = (props) => {
+  const verifyUser = () => {
+    props.setUsername("");
```
Comment (Melizzo) [inline src/Login/Login.tsx:14]: Reset for input upon logging into app.

File: src/Login/Login.tsx [hunk 3]
```diff
@@ -1,65 +1,75 @@
...
-						aria-label="name-input"
-						className="input"
-						placeholder="username"
-						name="username"
-						value={props.username}
-						onChange={(event) => props.setUsername(event.target.value)}
-						required
-					/>
-				</section>
+  const buttonsEnabled = props.username.trim() !== "";
```
Comment (Melizzo) [inline src/Login/Login.tsx:24]: Functionality that disables both buttons if the input is empty.  
trim() removes any whitespace at beginning or end of input that User may enter.  Whitespace is left if it is surrounded by characters, ie user name

File: src/Login/Login.tsx [hunk 4]
```diff
@@ -1,65 +1,75 @@
...
-		</section>
-	);
+        <section className="login-btns">
+          <Link to="/dashboard">
+            <button
+              onClick={(event) => verifyUser()}
+              className="submit-login-btn"
+              aria-label="submit-button"
+              type="button"
+              disabled={!buttonsEnabled}
```
Comment (Melizzo) [inline src/Login/Login.tsx:54]: Click for checking if buttons are enabled.

File: src/Login/Login.tsx [hunk 5]
```diff
@@ -1,65 +1,75 @@
...
+              I'm 21+
+            </button>
+          </Link>
+
+          <button
+            onClick={(event) => denyUser(event)}
+            className="submit-login-btn"
+            aria-label="submit-button"
+            type="button"
+            disabled={!buttonsEnabled}
```
Comment (Melizzo) [inline src/Login/Login.tsx:65]: Click for checking if buttons are enabled.

File: src/MyCocktails/MyCocktails.tsx [hunk 6]
```diff
@@ -1,9 +1,23 @@
...
+import { getAllCocktails } from '../apiCalls';
+// import './MyCocktails.scss';
+
 
 interface MyCocktailsProps {
 }
 
 const MyCocktails: React.SFC<MyCocktailsProps> = () => {
+	const [allCocktails, setAllCocktails] = useState({})
+
```
Comment (Melizzo) [inline src/MyCocktails/MyCocktails.tsx:11]: Notes for adding array of recipes, not functional at all, more of a mental roadmap.

File: src/Login/Login.tsx (hunk 1 above)
Comment (asodemann18) [reply src/Login/Login.tsx:7]: nice job fixing these!

Comment (st0nebreaker) [reply src/Login/Login.tsx:7]: ahhhh, i see. thanks for figuring that out! i was scratching my head about what to type setUsername as for some dumb reason hah

File: src/Login/Login.tsx (hunk 3 above)
Comment (asodemann18) [reply src/Login/Login.tsx:24]: Makes sense.

**Review by asodemann18 — APPROVED** [review]
Everything looks good!

**Comment by st0nebreaker** [comment]
Nice use of trim

**Pull request merged**